lib/shrine/plugins/blurhash.rb

```diff
@@ -90,10 +90,25 @@ def instrument_blurhash(io, &block)
 
       module InstanceMethods
         def extract_metadata(io, **options)
-          return super unless self.class.opts[:blurhash][:auto_extraction]
+          metadata = super
+
+          if self.class.opts[:blurhash][:auto_extraction] && image?(io, metadata)
+            begin
```

Comment:
I dont think this is useful anymore with the latest commit I pushed that reworks error handling